src/property/types/CloudSchedule.h

```diff
@@ -0,0 +1,417 @@
+#include "utility/time/TimeService.h"
+#include <time.h>
+
+/******************************************************************************
+ * DEFINE
+ ******************************************************************************/
+#define SCHEDULE_UNIT_MASK    0xC0000000
+#define SCHEDULE_UNIT_SHIFT   30
+
+#define SCHEDULE_TYPE_MASK    0x3C000000
+#define SCHEDULE_TYPE_SHIFT   26
+
+#define SCHEDULE_MONTH_MASK   0x0000FF00
+#define SCHEDULE_MONTH_SHIFT  8
+
+#define SCHEDULE_REP_MASK     0x03FFFFFF
+#define SCHEDULE_WEEK_MASK    0x000000FF
+#define SCHEDULE_DAY_MASK     0x000000FF
+
+#define SCHEDULE_ONE_SHOT     0xFFFFFFFF
+
+/******************************************************************************
+   ENUM
+ ******************************************************************************/
+enum class ScheduleUnit : int {
+  Seconds      = 0,
+  Minutes      = 1,
+  Hours        = 2,
+  Days         = 3
+};
+
+enum class ScheduleType : int {
+  OneShot      = 0,
+  FixedDelta   = 1,
+  Weekly       = 2,
+  Monthly      = 3,
+  Yearly       = 4
+};
+
+enum class ScheduleMonth : int {
+  Jan          = 0,
+  Feb          = 1,
+  Mar          = 2,
+  Apr          = 3,
+  May          = 4,
+  Jun          = 5,
+  Jul          = 6,
+  Aug          = 7,
+  Sep          = 8,
+  Oct          = 9,
+  Nov          = 10,
+  Dec          = 11
+};
+
+enum class ScheduleWeekDay : int {
+  Sun          = 0,
+  Mon          = 1,
+  Tue          = 2,
+  Wed          = 3,
+  Thu          = 4,
+  Fri          = 5,
+  Sat          = 6
+};
+
+enum class ScheduleState : int {
+  Inactive     = 0,
+  Active       = 1
+};
+
+/******************************************************************************
+ * TYPEDEF
+ ******************************************************************************/
+typedef struct ScheduleWeeklyMask {
+  ScheduleState& operator[](ScheduleWeekDay i) { return day[static_cast<int>(i)];}
+  ScheduleState day[7];
+}ScheduleWeeklyMask;
+
+/******************************************************************************
+   CLASS DECLARATION
+ ******************************************************************************/
+class Schedule : public TimeService {
+  public:
+    unsigned int frm, to, len, msk;
+    Schedule(unsigned int s, unsigned int e, unsigned int d, unsigned int m): frm(s), to(e), len(d), msk(m) {}
+
+    bool isActive() {
+
+      unsigned int now = getLocalTime();
+      if(checkSchedulePeriod(now, frm, to)) {
+        /* We are in the schedule range */
+
+        if(checkScheduleMask(now, msk)) {
+        
+          /* We can assume now that the schedule is always repeating with fixed delta */ 
+          unsigned int delta = getScheduleDelta(msk);
+          if ( ( (std::max(now , frm) - std::min(now , frm)) % delta ) <= len ) {
+            return true;
+          }
+        }
+      }
+      return false;
+    }
+
+    static unsigned int createOneShotScheduleConfiguration() {
+      return 0;
```

Comment:
what should this function do?

Reply:
for non repeating schedule the configurazion mask is 0

Reply:
oh right!